docs/topics/proc_cli-execute.adoc

```diff
@@ -25,56 +27,115 @@ $ ./mta-cli --input /path/to/jee-example-app-1.0.0.ear --output /path/to/output
 ----
-$ <MTA_HOME>/bin/mta-cli --sourceMode --input /path/to/seam-booking-5.2/ --output /path/to/report-output/ --target eap:6 --packages org.jboss.seam
+$ <MTA_HOME>/bin/mta-cli --sourceMode --input /path/to/seam-booking-5.2/ \
+--output /path/to/report-output/ --target eap:6 --packages org.jboss.seam
 ----
 
 [discrete]
 === Running cloud-readiness rules
 
 The following command analyzes the `com.acme` and `org.apache` packages of the link:https://github.com/windup/windup/blob/master/test-files/jee-example-app-1.0.0.ear[jee-example-app-1.0.0.ear] example EAR archive for migrating to JBoss EAP 7. It also evaluates for cloud readiness:
 
-[source,options="nowrap",subs="+quotes"]
+[source,terminal]
 ----
-$ <MTA_HOME>/bin/mta-cli --input /path/to/jee-example-app-1.0.0.ear --output /path/to/report-output/ --target eap:7 --target cloud-readiness --packages com.acme org.apache
+$ <MTA_HOME>/bin/mta-cli --input /path/to/jee-example-app-1.0.0.ear \
+ --output /path/to/report-output/ \
+ --target eap:7 --target cloud-readiness --packages com.acme org.apache
 ----
 
 [discrete]
 === Overriding {ProductShortName} properties
 
 To override the default _Fernflower_ decompiler, pass the `-Dwindup.decompiler` argument on the command line. For example, to use the _Procyon_ decompiler, use the following syntax:
 
-[source,options="nowrap",subs="+quotes"]
+[source,terminal]
 ----
-$ <MTA_HOME>/bin/mta-cli -Dwindup.decompiler=procyon --input <INPUT_ARCHIVE_OR_DIRECTORY> --output <OUTPUT_REPORT_DIRECTORY> --target <TARGET_TECHNOLOGY> --packages <PACKAGE_1> <PACKAGE_2>
+$ <MTA_HOME>/bin/mta-cli -Dwindup.decompiler=procyon --input \
+<INPUT_ARCHIVE_OR_DIRECTORY> --output <OUTPUT_REPORT_DIRECTORY> \
+--target <TARGET_TECHNOLOGY> --packages <PACKAGE_1> <PACKAGE_2>
 ----
+[id="openrewrite_{context}"]
+== Refactoring source code using OpenRewrite
+
+link:https://docs.openrewrite.org/[OpenRewrite] uses recipes to automate large-scale, distributed source code refactoring. It can be used with the {ProductShortName}.
+
+The first OpenRewrite recipe, which is shipped with {ProductShortName} 5.2.1, renames imported `javax` packages to their `jakarta` equivalents.
+
+You can use OpenRewrite in the {ProductShortName} CLI to prepare Java applications for migration.
+
+.Procedure
+
+. Add the `openrewrite` argument when you execute the 'mta-cli' script and specify the other arguments appropriately:
++
+[source,terminal]
+----
+./mta-cli --openrewrite
+"-DactiveRecipes=<recipe name>" --input
+/path/to/source/project --goal dryRun
+----
+
+* `--openrewrite`: Flag specifying to run an OpenRewrite migration, instead of an MTA analysis.
+
+* `"-DactiveRecipes=org.<recipe name>"` : The OpenRewrite recipe to apply to the input project. `JavaxtoJakarta` is the default shipped recipe but you can add your own to the shipped `rewrite.yml` and run that recipe instead. The `rewrite.yml file` is located in the `rules/openrewrite/`` folder in the uncompressed MTA distribution.
```

Comment:
org.<recipe name> should be just <recipe name>

Also as we are using a substitution variable in the parameter name it's probably a good idea to name the shipped recipe `org.jboss.windup.JavaxToJakarta` rather than just `JavaxToJakarta` in the description, as it won't work without the fully qualified name and the user will have no way of knowing it otherwise.